### ISSUE TITLE
Error handling while generating a JWT 

### ISSUE THREAD

**Issue by AmaliMatharaarachchi** [ballerina-platform/ballerina-lang#20896]
**Description:**

jwt:issueJWT supposed to return string|error as per the document https://ballerina.io/learn/api-docs/ballerina/jwt/functions.html#issueJwt. However it does not properly return an error. Instead it throws an error internally and exit. 

**Steps to reproduce:**
An example:
```
public function generateAPIKey() {
       crypto:KeyStore keyStore = {
            path: path,
            password: password
        };

        jwt:JwtKeyStoreConfig config = {
            keyStore: keyStore,
            keyAlias: wrong alias,
            keyPassword: password
        };
       ...
       ...
        printDebug(API_KEY_UTIL, "API Key is being issued.. .");
        string | error apiKey = jwt:issueJwt(header, jwtPayload, config);
        printDebug(API_KEY_UTIL, "API Key issuing process completed");
}
```
 if we provide an invalid alias it will exit with 
```
2020-02-06 17:15:13,212 DEBUG [APIKeyUtil] [-] API Key is being issued.. .
error: {ballerina/crypto}Error message=Unable to open keystore: keystore password was incorrect
	at ballerina:externDecodePrivateKey(crypto.bal:341)
	   ballerina:decodePrivateKey(crypto.bal:338)
	   ballerina.jwt:issueJwt(jwt_issuer.bal:70)
	   wso2.gateway.src.gateway.utils.api_key_utils:generateAPIKey(/src/gateway/utils/api_key_utils.bal:98)
	   wso2.pet.apiKeyService__service_4:apiKeyResource(tokenServices.bal:144)
```

Similarly, If we provide a wrong password as above, it will exit with
```
2020-02-06 17:15:13,212 DEBUG [APIKeyUtil] [-] API Key is being issued.. . 
error: java.lang.NullPointerException 
	at ballerina:externDecodePrivateKey(crypto.bal:65535)
	   ballerina:decodePrivateKey(crypto.bal:338)
	   ballerina.jwt:issueJwt(jwt_issuer.bal:70)
	   wso2.gateway.src.gateway.utils.api_key_utils:generateAPIKey(/src/gateway/utils/api_key_utils.bal:98)
	   wso2.pet.apiKeyService__service_4:apiKeyResource(tokenServices.bal:144)
```
Note that printDebug(API_KEY_UTIL, "API Key is being issued.. ."); printed while printDebug(API_KEY_UTIL, "API Key issuing process completed"); did not.

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by Rajith90** [comment]
We have checked this with baklerina 1.1.4 and it still fails with the exception[1]

[1] -
```java
error: java.lang.NullPointerException 
	at ballerina:externDecodePublicKey(crypto.bal:65535)
	   ballerina:decodePublicKey(crypto.bal:353)
	   ballerina.jwt:validateCertificate(jwt_validator.bal:285)
	   ballerina.jwt:validateJwtRecords(jwt_validator.bal:231)
	   ballerina.jwt:validateJwt(jwt_validator.bal:69)
	   ballerina.jwt.InboundJwtAuthProvider:authenticate(inbound_jwt_auth_provider.bal:57)
```

**Comment by ldclakmal** [comment]
@Rajith90 I tried following sample with Ballerina 1.1.4. It returns an error because of invalid key alias.

```ballerina
import ballerina/crypto;
import ballerina/jwt;
import ballerina/time;
import ballerina/io;

public function main() {
    crypto:KeyStore keyStore = { 
        path: "${ballerina.home}/bre/security/ballerinaKeystore.p12", 
        password: "ballerina"
    };
    jwt:JwtKeyStoreConfig config = {
        keyStore: keyStore,
        keyAlias: "invalid alias",
        keyPassword: "ballerina"
    };

    jwt:JwtHeader header = {};
    header.alg = jwt:RS256;
    header.typ = "JWT";

    jwt:JwtPayload payload = {};
    payload.sub = "John";
    payload.iss = "wso2";
    payload.jti = "100078234ba23";
    payload.aud = ["ballerina", "ballerinaSamples"];
    payload.exp = time:currentTime().time/1000 + 600;

    string|error jwt = jwt:issueJwt(header, payload, config);
    io:println(jwt);
}
```

**Comment by Rajith90** [comment]
Yes this NPE happens when invalid alias is provided. And the service will respond directly NPE to the client. Its better to send give a different error message or provide a way to give custom error message

**Comment by ldclakmal** [comment]
After the above fix it should return an Ballerina error instead or throwing java NPE. I checked the behavior with Ballerina 1.1.4 release. Do you still get the NPE?

**Comment by AmaliMatharaarachchi** [comment]
This issue happens for JWT validation as well. The issue is reproducible on Ballerina 1.1.4 version

**Comment by ldclakmal** [comment]
@AmaliMatharaarachchi This issue https://github.com/ballerina-platform/ballerina-lang/issues/20896 has only fixed the NPE thrown when generating the JWT as the original comment explains. That should be fixed by v1.1.4. Please verify.
We will fix the NPE thrown when validating the JWT as well, and update the ticket.

**Comment by AmaliMatharaarachchi** [comment]
@ldclakmal Created a new issue to track it. https://github.com/ballerina-platform/ballerina-lang/issues/22864
